crates/fluvio/src/producer/mod.rs

```diff
@@ -125,6 +175,46 @@ impl TopicProducer {
 
         Ok(())
     }
+
+    /// Flushes any buffered records to the SPU, waiting until the whole buffer has been emptied.
+    ///
+    /// This should be used when you want to be sure that records have been sent
+    /// before moving on to something else. If you are sending many records, you
+    /// should _first_ [`send`] all the records, _then_ call flush.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use fluvio::FluvioError;
+    /// # async fn example() -> Result<(), FluvioError> {
+    /// # let producer = fluvio::producer("topic").await?;
+    /// for i in 0..100 {
+    ///     producer.send(i.to_string(), format!("Hello, {}", i)).await?;
+    /// }
+    /// producer.flush().await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn flush(&self) -> Result<(), FluvioError> {
+        // Create a channel for the dispatcher to notify us when flushing is done
+        let (tx, rx) = async_channel::bounded(1);
```

Comment:
so what happens if you try to flush twice?

Reply:
Flushing twice works as expected. Here we have a unique `bounded(1)` for _each_ call to flush, so they do not interfere with one another.